analyzers/src/SonarAnalyzer.CSharp/Rules/ConditionalSimplification.cs

```diff
@@ -72,6 +75,24 @@ internal static bool TryGetExpressionComparedToNull(ExpressionSyntax expression,
                     return true;
                 }
             }
+
+            if (IsPatternExpressionSyntaxWrapper.IsInstance(expression.RemoveParentheses()))
+            {
+                var isPatternWrapper = (IsPatternExpressionSyntaxWrapper)expression.RemoveParentheses();
```

Comment:
Please add a case like:
Apple a = null;

Apple x;
if (a is (not null)) // Noncompliant {{Use the '??' operator here.}}
{
    x = a;
}
else
{
    x = b;
}

I assume that it will not work until you don`t rebase on master as there is the implementation for RemoveParentheses that removes  parenthesize for ParenthesizedPatternSyntax.

Reply:
Good point. Small note though: rebase will not solve this issue since we have to call `RemoveParentheses` also when handling inner patterns. 

One more note: we currently have a mess (code duplication) with methods named `RemoveParentheses` and the fix was applied in one place and not the other. We need to address this but I will do it in a separate PR after we split the code in multiple projects.